### PR TITLE
Add documenation and output selection to foreign toplevel example

### PR DESCRIPTION
**Pull request by fwSmit**
Made this for testing https://github.com/swaywm/sway/pull/5955.
While I was at it, I added some documentation as well.

### CODE REVIEW

**Comment by emersion** [comment]
I don't think we should have an option to print all outputs. Users can already use `wayland-info` to print them.

Please fix the indentation of these changes. We use tabs.

**Comment by fwSmit** [comment]
Sure, I fixed.

File: examples/foreign-toplevel.c [hunk 1]
```diff
@@ -48,6 +49,62 @@ struct toplevel_state {
...
 };
 
+struct wayland_output {
+	struct wl_list link;
+	uint32_t id;
+	struct wl_output *output;
+	char *make;
+	char *model;
+	int width;
+	int height;
```
Comment (ifreund) [inline examples/foreign-toplevel.c:59]: These are never used.

**Review by ifreund — COMMENTED** [review]
it might be simpler to handle this in the same way the layer-shell example client does instead of storing a list.

**Comment by fwSmit** [comment]
I guess I copied from the wrong example. The layer-shell example is much easier. I've implemented it that way now.

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Pull request merged**